sense-android-library/src/nl/sense_os/datastorageengine/CommonSenseProxy.java

```diff
@@ -0,0 +1,597 @@
+        JSONObject postData = new JSONObject();
+        postData.put("device", sensor);
+
+        String url = URL_SENSORS + "/" + csSensorID + "/" + URL_SENSOR_DEVICE;
+               url = makeUrlFor(url,null);
+        // perform actual request
+        Map<String, String> response = request(url, postData, sessionID, HTTP_METHOD_POST);
+        int code = checkResponseCode(response.get(RESPONSE_CODE), "addSensor");
+        // check if the response code is 201 CREATED
+        if(code != -2)
+            throw new IOException("Incorrect response of addSensor from CommonSense: " + code);
+        return (code == -2);
+    }
+
+    /**
+     * Upload sensor data to commonsense
+     *
+     * Data can be coming from multiple sensors.
+     *
+     * @param data			JSONArray of datapoints.
+     *                      Each datapoint should be a JSONObject with fields called "sensorID", "value", and "date".
+     *                      These fields will be parsed into the correct form for uploading to commonsense.
+     * @param sessionID		The sessionID of the current user. Cannot be empty.
+     * @throws IOException, JSONException
+     * @result				Whether or not the post of the data was succesfull.
+     */
+    public boolean postData(JSONArray data, String sessionID) throws IOException, JSONException
+    {
+        if(sessionID == null || sessionID.isEmpty() || data == null || data.length() == 0)
+            throw new IOException("invalid input of data or sessionID");
+
+        JSONObject postData = new JSONObject();
+        postData.put("sensors",data);
+
+        String url = makeUrlFor(URLE_UPLOAD_MULTIPLE_SENSORS,null);
+        Map<String, String> response = request(url, postData, sessionID,"POST");
+        int code = checkResponseCode(response.get(RESPONSE_CODE), "postData");
+        if(code != -2)
+            throw new IOException("Incorrect response of postData from CommonSense: " + response.get(RESPONSE_CODE));
+        return (code == -2);
+    }
+
+    /**
+     * Download sensor data from commonsense from a certain date till now.
+     *
+     * The downloaded data will be passed as an JSONArray to the success callback method from which it can be further processed.
+     *
+     * @param sensorID		Identifier of the sensor from CommonSense for which to download the data. Cannot be empty.
+     * @param fromDate	    Date from which to download data. Data points after this date will be included in the download.
+     *                          Data points before this date will be ignored. Cannot be null.
+     * @param sessionID		    The sessionID of the current user. Cannot be empty.
+     * @throws IOException, JSONException
+     * @result				    JSONArray with the resulting data. Each object is a JSONObject with the data as provided by the backend. Will be null if an error occurred.
+     */
+    public JSONArray getData(String sensorID, double fromDate, String sessionID) throws IOException, JSONException
```

Comment:
@excaliburnan see comments above about exception documentation and JSONException